### PR TITLE
Use common folder for all caches, make it a volume in Dockerfile

### PR DESCRIPTION
**Pull request by borzunov**

### CODE REVIEW

File: src/petals/utils/disk_cache.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import os
+from pathlib import Path
+
+
+DEFAULT_CACHE_DIR = os.getenv("PETALS_CACHE", Path(Path.home(), ".cache", "petals"))
```
Comment (mryab) [inline src/petals/utils/disk_cache.py:4]:
```suggestion
DEFAULT_CACHE_DIR = os.getenv("PETALS_CACHE", Path.home() / ".cache" / "petals")
```

File: src/petals/server/throughput.py [hunk 2]
```diff
@@ -15,31 +14,32 @@
...
 def get_host_throughput(
     config: BloomConfig,
     device: torch.device,
     dtype: Union[str, torch.dtype],
     *,
     load_in_8bit: bool,
     force_eval: bool = False,
-    cache_path: str = DEFAULT_CACHE_PATH,
-    lock_path: str = DEFAULT_LOCK_PATH,
+    cache_dir: Optional[str] = None,
```
Comment (mryab) [inline src/petals/server/throughput.py:30]:
```suggestion
    cache_dir: Optional[Union[str, Path]] = None,
```

**Review by mryab — APPROVED** [review]

File: src/petals/utils/disk_cache.py (hunk 1 above)
Comment (borzunov) [reply src/petals/utils/disk_cache.py:4]: I don't like that this notation doesn't work for all strings and less readable for Python amateurs

File: src/petals/server/throughput.py (hunk 2 above)
Comment (borzunov) [reply src/petals/server/throughput.py:30]: `Path` has all interfaces of `str` (duck typing), so I'll leave my variant to not overcomplicate the code

**Pull request merged**